### PR TITLE
[VK] BugFix for Deleting Pod

### PR DESCRIPTION
**Pull request by robbiezhang**
1. Fix the wrong error handling in deletePod. It should return error when delErr is **NOT** NotFound.
2. If delete failed, we don't retry. It's a potential cause of the pod leaking.

### CODE REVIEW

**Comment by robbiezhang** [comment]
Fix #400

**Comment by cpuguy83** [comment]
This looks pretty good, I'll run this through some tests.

File: vkubelet/pod.go [hunk 1]
```diff
@@ -179,30 +183,28 @@ func (s *Server) deletePod(ctx context.Context, pod *corev1.Pod) error {
 	addPodAttributes(span, pod)
 
 	var delErr error
-	if delErr = s.provider.DeletePod(ctx, pod); delErr != nil && errors.IsNotFound(delErr) {
+	if delErr = s.provider.DeletePod(ctx, pod); delErr != nil && !errors.IsNotFound(delErr) {
```
Comment (cpuguy83) [inline vkubelet/pod.go]: I don't think providers are returning a K8s error type? I don't think this is working as intended with or without the change.

We need providers to convert to errors that we can understand here.

Comment (cpuguy83) [reply vkubelet/pod.go]: We could use something like `strongerrors.NotFound(err)` in the provider and `strongerrors.IsNotFound(err)` here.

Would be nice to make sure all providers do this, but can fallback to `GetPod` as well.

Comment (robbiezhang) [reply vkubelet/pod.go]: Sounds good

Comment (cpuguy83) [reply vkubelet/pod.go]: https://github.com/virtual-kubelet/virtual-kubelet/pull/403 adds this to providers.

Comment (cpuguy83) [reply vkubelet/pod.go]: btw I think we want this to return when it is a NotFound error because there is nothing to do.

Comment (cpuguy83) [reply vkubelet/pod.go]: Or more specifically, we want to requeue with it's not a NotFound error.

Comment (robbiezhang) [reply vkubelet/pod.go]: we need to hard delete the pod from k8s

**Comment by robbiezhang** [comment]
thanks to #425 , this is no longer needed.

**Pull request closed without merging**